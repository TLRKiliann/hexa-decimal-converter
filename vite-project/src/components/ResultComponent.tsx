type ResultBinProps = {
  result: number | string | undefined;
}

export default function ResultComponent(props: ResultBinProps) {
  return (
    <>
      <span data-testid="span-testid">
        Result : {props.result ? props.result : "Not binary !!!"}
      </span>
    </>
  )
}
