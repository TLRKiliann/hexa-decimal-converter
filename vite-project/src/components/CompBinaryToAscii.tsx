import React, { useState } from 'react'
import LabelInputComponent from './LabelInputComponent.tsx'
import ResultComponent from './ResultComponent.tsx'
import ButtonComponent from './ButtonComponent.tsx';

function CompBinaryToAscii() {
    
    const [value, setValue] = useState<string>("");
    const [result, setResult] = useState<string | undefined>();

    const nameValue = "Binary";

    const converterTarget = (event: React.ChangeEvent<HTMLInputElement>) => {
        const etv = event?.target.value;
        setValue(etv);
    }
    
    const convertResult = () => {
        const parser = value.split(" ");
        const convertToAscii = parser.map((elem) => String.fromCharCode(parseInt(elem, 2)));
        const resltOfConversion = convertToAscii.join("");
        setResult(resltOfConversion);
        setValue("");
    }

    return (
        <>

            <div className="labelinput--box">
                <LabelInputComponent
                    value={value}
                    nameValue={nameValue}
                    converterTarget={(event)=>converterTarget(event)}
                >{nameValue}</LabelInputComponent>
            </div>

            <ButtonComponent convertResult={convertResult} />
            
            <div className="box--result">
                <ResultComponent 
                    result={result}
                />
            </div>

        </>
    )
}

export default CompBinaryToAscii