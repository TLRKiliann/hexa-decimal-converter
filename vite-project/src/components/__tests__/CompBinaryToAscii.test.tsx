import { test, expect, vi } from "vitest"
import { render } from '@testing-library/react'
import { create } from "react-test-renderer"
import "@testing-library/jest-dom"
import CompBinaryToAscii from '../CompBinaryToAscii'
import ResultComponent from "../ResultComponent"
//import LabelInputComponent from "../LabelInputComponent"
//import { ChangeEvent } from "react";

test('Snapshat of CompBinaryToAscii', () => {
    const tree = create(<CompBinaryToAscii />)
    expect(tree.toJSON()).toMatchSnapshot()
});

test('convertResult return yeah', () => {
    const beverage = "yeah";
    const convertResult = vi.fn(beverage => beverage)
    convertResult(beverage)
    render(
        <ResultComponent result={"yeah"} />
    );
    //fireEvent.click(screen.getByTestId('testid-btn'))
    expect(convertResult).toHaveReturnedWith("yeah")
    expect(convertResult).toHaveBeenCalledTimes(1)
});
/*
describe("captures event from FirstComp", async () => {
    test('testing function handleEventChange', () => {
        function handleEventChange(evt: ChangeEvent<HTMLInputElement>) {
        expect(evt.target.value).toEqual("changedvalue");
        }
        render(
        <LabelInputComponent value={undefined} nameValue={""}>
            <input
            data-testid="testid-event"
            type="text"
            value=""
            onChange={(evt) => handleEventChange(evt)}
            />
        </LabelInputComponent>
        )
        //const node = screen.getByTestId("testid-input");
        //fireEvent.change(node, { target: { value: "changedvalue" }});
        expect(handleEventChange).toBeDefined();
    })
});
*/