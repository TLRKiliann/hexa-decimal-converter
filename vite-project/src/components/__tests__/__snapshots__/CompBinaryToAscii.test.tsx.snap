// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshat of CompBinaryToAscii 1`] = `
[
  <div
    className="labelinput--box"
  >
    <label>
      Binary
       :
    </label>
    <input
      data-testid="testid--input"
      onChange={[Function]}
      placeholder="Binary"
      type="text"
      value=""
    />
  </div>,
  <button
    data-testid="testid-btn"
    onClick={[Function]}
    type="button"
  >
    Convert
  </button>,
  <div
    className="box--result"
  >
    <span
      data-testid="span-testid"
    >
      Result : 
      Not binary !!!
    </span>
  </div>,
]
`;
